### ISSUE TITLE
Panic when cutting a new TSDB chunk file

### ISSUE THREAD

**Issue by juliusv** [prometheus/prometheus#10769]
### What did you do?

I ran Prometheus scraping itself and using remote write+read (with a local Cortex) as in the config below. The storage dir was small (~11MB), but not completely fresh.

The Prometheus server is a local build off of a 2.33.0-rc.0-ish branch without local TSDB changes, but not 100% sure whether the bug was fixed by now.

### What did you expect to see?

Everything working fine, no crash.

### What did you see instead? Under which circumstances?

Prometheus crashed with a panic:

```yaml
panic: expected newly cut file to have sequence:offset 2:8, got 1:8
```

Full logs below.

### System information

Linux 5.11.0-49-generic x86_64

### Prometheus version

```text
prometheus, version 2.33.0-rc.0 (branch: bolden-graph-lines, revision: a979e8311f59340b762cd168c6c892b3ab12377e)
  build user:       julius@julius-dell
  build date:       20220119-11:13:41
  go version:       go1.17.5
  platform:         linux/amd64
```

### Prometheus configuration file

```yaml
global:
  scrape_interval: 5s

# Write data to Cortex.
remote_write:
  - url: "http://localhost:9009/api/prom/push"
# Read data from Cortex.
remote_read:
  - url: "http://localhost:9009/api/prom/api/v1/read"

# Let Prometheus scrape itself.
scrape_configs:
  - job_name: 'prometheus'
    static_configs:
      - targets:
        - 'localhost:9090'
```

### Logs

```text
./prometheus --config.file=rw-training.yml
ts=2022-05-28T08:43:40.796Z caller=main.go:475 level=info msg="No time or size retention was set so using the default time retention" duration=15d
ts=2022-05-28T08:43:40.796Z caller=main.go:512 level=info msg="Starting Prometheus" version="(version=2.33.0-rc.0, branch=bolden-graph-lines, revision=a979e8311f59340b762cd168c6c892b3ab12377e)"
ts=2022-05-28T08:43:40.796Z caller=main.go:517 level=info build_context="(go=go1.17.5, user=julius@julius-dell, date=20220119-11:13:41)"
ts=2022-05-28T08:43:40.796Z caller=main.go:518 level=info host_details="(Linux 5.11.0-49-generic #55-Ubuntu SMP Wed Jan 12 17:36:34 UTC 2022 x86_64 julius-dell (none))"
ts=2022-05-28T08:43:40.796Z caller=main.go:519 level=info fd_limits="(soft=1024, hard=1048576)"
ts=2022-05-28T08:43:40.796Z caller=main.go:520 level=info vm_limits="(soft=unlimited, hard=unlimited)"
ts=2022-05-28T08:43:40.798Z caller=web.go:570 level=info component=web msg="Start listening for connections" address=0.0.0.0:9090
ts=2022-05-28T08:43:40.798Z caller=main.go:923 level=info msg="Starting TSDB ..."
ts=2022-05-28T08:43:40.799Z caller=tls_config.go:195 level=info component=web msg="TLS is disabled." http2=false
ts=2022-05-28T08:43:40.800Z caller=repair.go:57 level=info component=tsdb msg="Found healthy block" mint=1639046188326 maxt=1639051200000 ulid=01FSS0MK6XDPV9XSJKVQ2PY21Z
ts=2022-05-28T08:43:40.803Z caller=head.go:493 level=info component=tsdb msg="Replaying on-disk memory mappable chunks if any"
ts=2022-05-28T08:43:40.806Z caller=head.go:527 level=info component=tsdb msg="On-disk memory mappable chunks replay completed" duration=2.776344ms
ts=2022-05-28T08:43:40.806Z caller=head.go:533 level=info component=tsdb msg="Replaying WAL, this may take a while"
ts=2022-05-28T08:43:40.992Z caller=head.go:569 level=info component=tsdb msg="WAL checkpoint loaded"
ts=2022-05-28T08:43:41.080Z caller=head.go:604 level=info component=tsdb msg="WAL segment loaded" segment=2 maxSegment=5
ts=2022-05-28T08:43:41.081Z caller=head.go:604 level=info component=tsdb msg="WAL segment loaded" segment=3 maxSegment=5
ts=2022-05-28T08:43:41.130Z caller=head.go:604 level=info component=tsdb msg="WAL segment loaded" segment=4 maxSegment=5
ts=2022-05-28T08:43:41.130Z caller=head.go:604 level=info component=tsdb msg="WAL segment loaded" segment=5 maxSegment=5
ts=2022-05-28T08:43:41.130Z caller=head.go:610 level=info component=tsdb msg="WAL replay completed" checkpoint_replay_duration=185.479963ms wal_replay_duration=138.138893ms total_replay_duration=326.412836ms
ts=2022-05-28T08:43:41.132Z caller=main.go:944 level=info fs_type=EXT4_SUPER_MAGIC
ts=2022-05-28T08:43:41.132Z caller=main.go:947 level=info msg="TSDB started"
ts=2022-05-28T08:43:41.132Z caller=main.go:1128 level=info msg="Loading configuration file" filename=rw-training.yml
ts=2022-05-28T08:43:41.133Z caller=dedupe.go:112 component=remote level=info remote_name=9d54c8 url=http://localhost:9009/api/prom/push msg="Starting WAL watcher" queue=9d54c8
ts=2022-05-28T08:43:41.133Z caller=dedupe.go:112 component=remote level=info remote_name=9d54c8 url=http://localhost:9009/api/prom/push msg="Starting scraped metadata watcher"
ts=2022-05-28T08:43:41.133Z caller=dedupe.go:112 component=remote level=info remote_name=9d54c8 url=http://localhost:9009/api/prom/push msg="Replaying WAL" queue=9d54c8
ts=2022-05-28T08:43:41.133Z caller=main.go:1165 level=info msg="Completed loading of configuration file" filename=rw-training.yml totalDuration=927.573µs db_storage=849ns remote_storage=320.456µs web_handler=288ns query_engine=445ns scrape=313.406µs scrape_sd=13.836µs notify=470ns notify_sd=8.849µs rules=985ns
ts=2022-05-28T08:43:41.133Z caller=main.go:896 level=info msg="Server is ready to receive web requests."
ts=2022-05-28T08:43:50.608Z caller=dedupe.go:112 component=remote level=info remote_name=9d54c8 url=http://localhost:9009/api/prom/push msg="Done replaying WAL" duration=9.475314592s
ts=2022-05-28T08:43:50.610Z caller=compact.go:510 level=info component=tsdb msg="write block resulted in empty block" mint=1639051200000 maxt=1639058400000 duration=12.664926ms
ts=2022-05-28T08:43:50.612Z caller=head.go:818 level=info component=tsdb msg="Head GC completed" duration=1.867008ms
ts=2022-05-28T08:43:50.649Z caller=compact.go:519 level=info component=tsdb msg="write block" mint=1642590784387 maxt=1642593600000 ulid=01G44XC1YMA66NSWFK6S414EQ6 duration=36.718949ms
ts=2022-05-28T08:43:50.651Z caller=db.go:1287 level=info component=tsdb msg="Deleting obsolete block" block=01FSS0MK6XDPV9XSJKVQ2PY21Z
ts=2022-05-28T08:43:50.652Z caller=head.go:818 level=info component=tsdb msg="Head GC completed" duration=882.31µs
ts=2022-05-28T08:43:50.695Z caller=compact.go:519 level=info component=tsdb msg="write block" mint=1642593601281 maxt=1642600800000 ulid=01G44XC1ZW5YJ4EQB3J33X1A6H duration=43.175138ms
ts=2022-05-28T08:43:50.699Z caller=head.go:818 level=info component=tsdb msg="Head GC completed" duration=3.067836ms
ts=2022-05-28T08:43:50.700Z caller=checkpoint.go:98 level=info component=tsdb msg="Creating checkpoint" from_segment=2 to_segment=3 mint=1642600800000
ts=2022-05-28T08:43:50.712Z caller=head.go:987 level=info component=tsdb msg="WAL checkpoint complete" first=2 last=3 duration=12.819902ms
panic: expected newly cut file to have sequence:offset 2:8, got 1:8

goroutine 440 [running]:
github.com/prometheus/prometheus/tsdb.handleChunkWriteError({0x35e5440, 0xc0044165a0})
	/home/julius/gosrc/src/github.com/prometheus/prometheus/tsdb/head_append.go:585 +0x76
github.com/prometheus/prometheus/tsdb/chunks.(*chunkWriteQueue).processJob(0xc0008c4980, {0x1, 0x8fb, 0x18109d607b9, 0x18109dff2c9, {0x365fc18, 0xc000d98b00}, 0x200000008, 0x3218db0})
	/home/julius/gosrc/src/github.com/prometheus/prometheus/tsdb/chunks/chunk_write_queue.go:106 +0xc2
github.com/prometheus/prometheus/tsdb/chunks.(*chunkWriteQueue).start.func1()
	/home/julius/gosrc/src/github.com/prometheus/prometheus/tsdb/chunks/chunk_write_queue.go:94 +0x165
created by github.com/prometheus/prometheus/tsdb/chunks.(*chunkWriteQueue).start
	/home/julius/gosrc/src/github.com/prometheus/prometheus/tsdb/chunks/chunk_write_queue.go:90 +0x72
```
```

**Comment by juliusv** [comment]
Attaching a tarball of the storage dir this happened with.

[data.tar.gz](https://github.com/prometheus/prometheus/files/8790885/data.tar.gz)

**Comment by roidelapluie** [comment]
Closed by #10316, fixed in v2.33.4

**Comment by juliusv** [comment]
Whoopsie, didn't spot that one, thanks and sorry for the noise!
